shared/ssl.sh

```diff
@@ -100,9 +100,11 @@ openssl genrsa -out "${PRIVKEYFILE}" 4096
 openssl rsa -in "${PRIVKEYFILE}" -pubout -out "${PUBKEYFILE}"
 openssl req -new -key "${PRIVKEYFILE}" -out "${CSRFILE}" -subj "${CERTSUBJECT}"
 
-### Submit CSR
-curl ${CURLFLAGS} -X PUT -H "Content-Type: text/plain" \
-    --data-binary "@${CSRFILE}" "${CA}/certificate_request/${CERTNAME}"
+### Submit CSR and fail if one already exists on the CA
+output="$(curl ${CURLFLAGS} -X PUT -H "Content-Type: text/plain" \
+               --data-binary "@${CSRFILE}" "${CA}/certificate_request/${CERTNAME}")"
+already_exists="${CERTNAME} already has a requested certificate; ignoring certificate request"
```

Comment:
@joshcooper would it be possible to use a HTTP status code rather than matching this text string?

I know you're working on some of the SSL negotiation now (not sure if just on the client side or server-side as well)

Reply:
@Magisus might be able to answer this question as well

Reply:
The server replies with 400 if it already has a CSR or cert with that name, even if it already has the same CSR that the client is sending. We talked about changing response codes, eg 409 means we already have a different CSR with that name, but we didn't want to introduce API incompatibility.

Reply:
Ok, so I'm not sure if `Bad Request` is specific enough / used only for this specific failure mode - so that means our best bet is probably what @nwolfe has done here.